CMakeLists.txt

```diff
@@ -138,103 +138,154 @@ set(XEUS_SOURCES
+            target_compile_options(${target_name} PUBLIC -mtune=generic)
+        else ()
+            target_compile_options(${target_name} PUBLIC -march=native)
+        endif ()
+
+        # Enable link time optimization and set the default symbol
+        # visibility to hidden (very important to obtain small binaries)
+        if (NOT ${U_CMAKE_BUILD_TYPE} MATCHES DEBUG)
+            # Check for Link Time Optimization support
+            # (GCC/Clang)
+            CHECK_CXX_COMPILER_FLAG("-flto" HAS_LTO_FLAG)
+            if (HAS_LTO_FLAG)
+                target_compile_options(${target_name} PUBLIC -flto)
+            endif ()
+
+            # Intel equivalent to LTO is called IPO
+            if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+                CHECK_CXX_COMPILER_FLAG("-ipo" HAS_IPO_FLAG)
+                if (HAS_IPO_FLAG)
+                    target_compile_options(${target_name} PUBLIC -ipo)
+                endif ()
+            endif ()
+        endif ()
+
+        message(STATUS "CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
     endif()
 
-    # Enable link time optimization and set the default symbol
-    # visibility to hidden (very important to obtain small binaries)
-    if (NOT ${U_CMAKE_BUILD_TYPE} MATCHES DEBUG)
-        # Check for Link Time Optimization support
-        # (GCC/Clang)
-        CHECK_CXX_COMPILER_FLAG("-flto" HAS_LTO_FLAG)
-        if (HAS_LTO_FLAG)
-            target_compile_options(xeus PUBLIC -flto)
-        endif()
-
-        # Intel equivalent to LTO is called IPO
-        if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-            CHECK_CXX_COMPILER_FLAG("-ipo" HAS_IPO_FLAG)
-            if (HAS_IPO_FLAG)
-                target_compile_options(xeus PUBLIC -ipo)
-            endif()
-        endif()
-    endif()
-    message(STATUS "CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
-endif()
+    if (${linkage_upper} STREQUAL "STATIC")
+        target_compile_definitions(${target_name} PUBLIC XEUS_STATIC_LIB)
+    endif ()
+
+    if (MSVC)
+        target_compile_definitions(${target_name} PUBLIC -DNOMINMAX)
+        target_compile_options(${target_name} PUBLIC /DGUID_WINDOWS /MP /bigobj)
+        target_compile_options(${target_name} PUBLIC /wd4251 /wd4996)
+    elseif (APPLE)
+        target_compile_definitions(${target_name} PUBLIC -DGUID_CFUUID)
+    else ()
+        target_compile_definitions(${target_name} PUBLIC -DGUID_LIBUUID)
+    endif ()
+endmacro()
+
+set(xeus_targets "")
+
+if (BUILD_SHARED_LIBS)
+    xeus_create_target(xeus "SHARED")
+    list(APPEND xeus_targets xeus)
+endif ()
 
-if(MSVC)
-    target_compile_definitions(xeus PUBLIC -DNOMINMAX)
-    target_compile_options(xeus PUBLIC /DGUID_WINDOWS /MP /bigobj)
-    target_compile_options(xeus PUBLIC /wd4251 /wd4996)
-elseif(APPLE)
-    target_compile_definitions(xeus PUBLIC -DGUID_CFUUID)
-else()
-    target_compile_definitions(xeus PUBLIC -DGUID_LIBUUID)
-endif()
+if (BUILD_STATIC_LIBS)
+    xeus_create_target(xeus_static "STATIC")
```

Comment:
Maybe set the OUTPUT_NAME to `xeus` on linux.

On windows, it may be different, because of the conflict with the import library. I don't know if there is a convention for the name of a static build on windows.